### ISSUE TITLE
Outfits are not going to stock when sold

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#1741]
Buy a brand new ship that contains a few outfits not sold at that planet's outfitter and then sell them; for me, those outfits are disappearing from the outfitter instead of going to stock.

**Comment by endless-sky** [comment]
That's to fix the exploit where you could buy a ship and then sell it back immediately, and repeat that any number of times, to generate an infinite stock of outfits not for sale on a given planet.

**Comment by Amazinite** [comment]
Fair enough.

**Comment by Bladewood** [comment]
*starts a new save*
*buys a Star Barge*
*sells Chipmunk engines*
*Chipmunk engines no longer exist*
waitwut
*remembers that this issue existed*

I think 0.9.5 is going to confuse many players...

**Comment by KiLEdEnNis** [comment]
wait, as soon as you sell them they disappear? even if you still own the ship?
if that's true, then if you swap stuff from ships you own you lose it
(or does it disappear after leaving the outfitter?)

**Comment by Amazinite** [comment]
Buy a brand new ship
Sell one of the outfits that isn't normally sold at the outfitter
It will instantly disappear and be unable to be bought back.

**Comment by KiLEdEnNis** [comment]
if you sell an outfit on a ship you already own it doesn't happen right?

**Comment by Amazinite** [comment]
If the outfit has any depreciation on it, selling it will put it into stock, but outfits with no depreciation that are sold will disappear, it seems. So yes, selling outfits from older ships doesn't make this happen.

**Comment by endless-sky** [comment]
The behavior was changed to fix the following exploit:
* An outfit is not sold here, but a ship is sold here that has that outfit in its stock configuration.
* You repeatedly buy and sell one (or more) of those ships.
* Now a large number of those outfits are "in stock."

The "fix" was to allow the "in stock" value to go negative when you buy a ship. That is, the outfitter remembers that that outfit is "borrowed" from somewhere, and won't make it available in stock if you sell the whole ship. But, that causes this issue.

In commit 0f4296c I've added a compromise:
* If you sell an individual outfit in the outfitter, it always will be "in stock" even if it was part of a ship you just bought here today.
* If you sell an entire ship, its outfits do not become "in stock" unless you've sold more than you indirectly bought by buying ships that contained them.
